x-pack/plugins/cases/server/client/cases/update_custom_field.ts

```diff
@@ -0,0 +1,177 @@
+
+    const caseToUpdate = await caseService.getCase({
+      id: caseId,
+    });
+
+    const configurations = await casesClient.configure.get({
+      owner: caseToUpdate.attributes.owner,
+    });
+
+    await authorization.ensureAuthorized({
+      entities: [{ owner: caseToUpdate.attributes.owner, id: caseToUpdate.id }],
+      operation: Operations.updateCase,
+    });
+
+    const foundCustomField = configurations[0]?.customFields.find(
+      (item) => item.key === customFieldId
+    );
+
+    if (!foundCustomField) {
+      throw Boom.badRequest('cannot find custom field');
+    }
+
+    const decodedCustomFields = decodeWithExcessOrThrow(CustomFieldsRt)([
+      {
+        value,
+        type: foundCustomField.type,
+        key: customFieldId,
+      },
+      ...caseToUpdate.attributes.customFields.filter((field) => field.key !== customFieldId),
+    ]);
+
+    const updatedAt = new Date().toISOString();
+
+    if (value == null && foundCustomField.required) {
+      throw Boom.badRequest('Custom field value cannot be null or undefined.');
+    }
+
+    validateCustomFieldKeysAgainstConfiguration({
+      requestCustomFields: decodedCustomFields,
+      customFieldsConfiguration: configurations[0].customFields,
+    });
+
+    validateCustomFieldTypesInRequest({
+      requestCustomFields: decodedCustomFields,
+      customFieldsConfiguration: configurations[0].customFields,
+    });
+
+    const customFieldsToUpdate: CaseRequestCustomFields = caseToUpdate.attributes.customFields.map(
+      (cf) => {
+        if (cf.key === customFieldId) {
+          return {
+            value,
+            type: foundCustomField.type,
+            key: customFieldId,
+          } as CaseRequestCustomField;
+        }
+        return cf;
+      }
+    );
+
+    const patchCasesPayload = {
+      caseId,
+      originalCase: caseToUpdate,
+      updatedAttributes: {
+        customFields: customFieldsToUpdate,
+        updated_at: updatedAt,
+        updated_by: user,
+      },
+      version: caseVersion,
+    };
+
+    const updatedCase = await caseService.patchCase({
+      ...patchCasesPayload,
+      refresh: false,
+    });
+
+    const userActionsDict = userActionService.creator.buildUserActions({
+      updatedCases: {
+        cases: [patchCasesPayload],
+      },
+      user,
+    });
+
+    const returnUpdatedCase = flattenCaseSavedObject({
+      savedObject: {
+        ...caseToUpdate,
+        ...updatedCase,
+        attributes: { ...caseToUpdate.attributes, ...updatedCase?.attributes },
+        references: updatedCase.references ?? caseToUpdate.references,
+        version: updatedCase?.version ?? caseToUpdate.version,
+      },
+    });
+
+    const builtUserActions =
+      userActionsDict != null
+        ? Object.keys(userActionsDict).reduce<UserActionEvent[]>((acc, key) => {
+            return [...acc, ...userActionsDict[key]];
+          }, [])
+        : [];
+
+    await userActionService.creator.bulkCreateUpdateCase({
```

Comment:
We need to throw if the max user actions are reached 😬

Reply:
ohh.. missed that 😓

Reply:
added fix and test in [bbd5d0c](https://github.com/elastic/kibana/pull/176168/commits/bbd5d0c2ee715995674a080f9a852c99ad148971)